### PR TITLE
Right toc and awards hero

### LINKED ISSUE #93: Right side TOC and awards hero
image

https://www.sunstar-foundation.org/en/awards/aap

### PR DESCRIPTION
**Pull request by JiangLong2019**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fixes #93 

Main change:

1. Refactored TOC block to support awards right side TOC
2. Introduced a new variant `image-right` to hero-banner block

Test URLs:
- Original: https://www.sunstar-foundation.org/en/awards/aap
- Before: https://main--sunstar-foundation--hlxsites.hlx.live/en/awards/aap
- After: https://right-toc-hero--sunstar-foundation--hlxsites.hlx.live/en/awards/aap

- [ ] New variations introduced in this PR

Variation Name    | Documentation
------------- | -------------
Hero Banner(image-right)   | [doc-link](https://right-toc-hero--sunstar-foundation--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/sidekick/blocks/hero-banner&index=7)

### CODE REVIEW

**Comment by RitwikSrivastava** [comment]
1. On clicking on right side TOC, it opens on new page. It should appear on the same page . 
2. On original site in mobile view, the TOC shifts to the bottom as shown below. 
Screenshot 2024-01-15 at 11 15 48 AM

File: blocks/toc/toc.css [hunk 1]
```diff
@@ -40,19 +31,121 @@ main .toc.flat.block ul li {
...
+main .toc.block ul {
+    padding: unset;
+    padding-bottom: 2rem;
+    margin: unset;
+    list-style: none;
+}
+
+main .toc.block ul li.active a::before {
+    content: "";
+    background: #007CAD;
```
Comment (RitwikSrivastava) [inline blocks/toc/toc.css:94]: Can we use the preset color here ?

File: blocks/toc/toc.css [hunk 2]
```diff
@@ -91,33 +184,64 @@ main .section:not([id='']) {
...
+    
+    .main-content .right-toc {
+        padding: 6.5625rem 0 0 2.8125rem;
+        position: absolute;
+        height: 100%;
+        top: 0;
+        right: 0;
+        width: calc(50% - 17.5rem);
+        order: 2;
+        background: #F0F5F7;
```
Comment (RitwikSrivastava) [inline blocks/toc/toc.css:199]: Same here.

File: blocks/toc/toc.css (hunk 1 above)
Comment (JiangLong2019) [reply blocks/toc/toc.css]: @RitwikSrivastava thanks, fixed.

File: blocks/toc/toc.css (hunk 2 above)
Comment (JiangLong2019) [reply blocks/toc/toc.css]: Fixed

**Comment by JiangLong2019** [comment]
@RitwikSrivastava 
>On clicking on right side TOC, it opens on new page. It should appear on the same page .

Fixed, please check.

>On original site in mobile view, the TOC shifts to the bottom as shown below. (We can confirm with Dejan if this behaviour is required)

I couldn't reproduce it as you described, the dropdown list always show after hero in mobile view. Can you tell me how did you reproduce it?

image

**Comment by RitwikSrivastava** [comment]
@JiangLong2019 The TOC is also there at the **bottom of the page** in mobile view in original site (which is kind of reduntant because the options are there below the hero too)

**Comment by JiangLong2019** [comment]
@RitwikSrivastava I see, let me bring this question to Dejan tomorrow. Removing the desktop one would be great for mobile view.

File: blocks/hero-banner/hero-banner.css [hunk 3]
```diff
@@ -205,7 +215,7 @@
         position: absolute;
         top: 28rem;
     }
-
+
```
Comment (sdmcraft) [inline blocks/hero-banner/hero-banner.css:218]: Looks like some unintentional spaces.

File: blocks/hero-banner/hero-banner.css [hunk 4]
```diff
@@ -258,6 +268,63 @@
         max-width: 56.5rem;
     }
 
+    .hero-banner-wrapper:has(.hero-banner.image-right) {
```
Comment (sdmcraft) [inline blocks/hero-banner/hero-banner.css:271]: I think `has` doesn't work with Firefox.

Comment (JiangLong2019) [reply blocks/hero-banner/hero-banner.css:271]: Yeah, that's in the past. Now FF supports it ^ ^
image

https://caniuse.com/css-has

**Comment by JiangLong2019** [comment]
Note: Confirmed with Dejan that we don't need to implement bottom TOC for mobile view.

**Comment by JiangLong2019** [comment]
@sdmcraft @RitwikSrivastava I think I addressed all feedback, can you please check? If it's ok, please help to approve it.

**Review by RitwikSrivastava — APPROVED** [review]

**Pull request merged**